raiden_contracts/contracts/MonitoringService.sol

```diff
@@ -0,0 +1,298 @@
+    mapping(address => uint) public balances;
+
+    /*
+     *  Structs
+     */
+    struct Reward{
+        // The amount of tokens to be rewarded
+        uint256 reward_amount;
+
+        // Nonce of the most recently provided BP
+        uint256 nonce;
+
+        // Address of the Raiden Node that was monitored
+        // This is also the address that has the reward deducted from its deposit
+        address reward_sender_address;
+
+        // Address of the Monitoring Service who is currenly eligible to claim the reward
+        address monitoring_service_address;
+    }
+
+    /*
+     *  Events
+     */
+
+    event NewDeposit(address receiver, uint amount);
+    event NewBalanceProofReceived(
+        uint256 reward_amount,
+        uint256 nonce,
+        address ms_address,
+        address raiden_node_address
+    );
+    event RewardClaimed(address ms_address, uint amount, bytes32 reward_identifier);
+    event Withdrawn(address account, uint amount);
+
+    /*
+     *  Modifiers 
+     */
+
+    modifier canMonitor(address _ms_address) {
+        require(rsb.deposits(_ms_address) > 0);
+        _;
+    }
+
+    /*
+     *  Constructor
+     */
+
+    /// @notice Set the default values for the smart contract
+    /// @param _token_address The address of the token to use for rewards
+    /// @param _rsb_address The address of the RaidenServiceBundle contract
+    constructor(
+        address _token_address,
+        address _rsb_address
+    )
+        public
+    {
+        require(_token_address != 0x0);
+        require(_rsb_address != 0x0);
+        require(contractExists(_token_address));
+        require(contractExists(_rsb_address));
+
+        token = Token(_token_address);
+        rsb = RaidenServiceBundle(_rsb_address);
+        // Check if the contract is indeed a token contract
+        require(token.totalSupply() > 0);
+        // Check if the contract is indeed an rsb contract
+        // TODO: Check that some function exists in the contract
+    }
+
+    /// @notice Deposit tokens used to either pay MSs or to regsiter as MS
+    /// Can be called by anyone several times and on behalf of other accounts
+    /// @param beneficiary The account benefitting from the deposit
+    /// @param amount The amount of tokens to be depositted
+    function deposit(address beneficiary, uint amount) public {
+        require(amount > 0);
+
+        // This also allows for MSs to deposit and use other MSs
+        balances[beneficiary] += amount;
+
+        // Transfer the deposit to the smart contract
+        require(token.transferFrom(msg.sender, address(this), amount));
+
+        emit NewDeposit(beneficiary, amount);
+    }
+
+    /// @notice Internal function that updates the Reward struct if a newer balance proof
+    /// is provided in the monitor() function
+    /// @param token_network_address Address of the TokenNetwork being monitored
+    /// @param closing_participant The address of the closing participant
```

Comment:
Something like `Address of the participant who closed the channel` may be clearer.